inc/Config/QueryContext/HttpQueryContext.php

```diff
@@ -180,4 +242,49 @@ public function process_response( string $raw_response_data, array $input_variab
 	final public function is_response_data_collection(): bool {
 		return $this->output_schema['is_collection'] ?? false;
 	}
+
+	// @todo: consider splitting the datasource injection out from query context so we don't have to tie a query
+	// to a datasource when instantiating. instead, we can just require applying queries to datasources in query
+	// runner execution. ie: $query_runner->execute( $query, $datasource );
```

Comment:
I don't think this bodes well for type safety, especially for implementors who provide custom datasources (because they have additional reusable properties they want to inject)

Reply:
Yeah, I'm not saying that's the only way or the best way. I just wanted to make a note because I'm feeling the tension in the current coupling and that thought came to mind.